### PR TITLE
changing term query to match query

### PR DESCRIPTION
**Pull request by adisri2694**
## JIRA Ticket Number

JIRA TICKET: ML-2964

## Description of change
Term query used in elastic search query needs to be converted to match query to support text field search.
## Checklist (OPTIONAL):

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datastore/elastic_search/query.py [hunk 1]
```diff
@@ -598,13 +598,18 @@ def get_crf_data_for_entity_name(connection, index_name, doc_type, entity_name,
...
-
-    if languages:
-        data['query']['bool']['filter'] = {
-            "terms": {
-                "language_script": languages
+    #       match query does not support array(list of languages).
+    #       building a list of dictionaries corresponding to match query for each language
+    list_query_langs = []
+    for lang in languages:
+        query_lang={
```
Comment (haptik-deployment) [inline datastore/elastic_search/query.py:605]: E225 missing whitespace around operator

File: datastore/elastic_search/query.py [hunk 2]
```diff
@@ -598,13 +598,18 @@ def get_crf_data_for_entity_name(connection, index_name, doc_type, entity_name,
...
+            "match": {
+                "language_script": i
```
Comment (haptik-deployment) [inline datastore/elastic_search/query.py:607]: F821 undefined name 'i'

File: datastore/elastic_search/query.py [hunk 3]
```diff
@@ -598,13 +598,18 @@ def get_crf_data_for_entity_name(connection, index_name, doc_type, entity_name,
...
             }
         }
+        list_query_langs.append(query_lang)
+
+    data['query']['bool']['should'] =list_query_langs
```
Comment (haptik-deployment) [inline datastore/elastic_search/query.py:612]: E225 missing whitespace around operator

File: datastore/elastic_search/query.py [hunk 4]
```diff
@@ -598,13 +598,18 @@ def get_crf_data_for_entity_name(connection, index_name, doc_type, entity_name,
...
         }
     }
-
-    if languages:
-        data['query']['bool']['filter'] = {
-            "terms": {
-                "language_script": languages
+    #       match query does not support array(list of languages).
+    #       building a list of dictionaries corresponding to match query for each language
+    if languages :
```
Comment (haptik-deployment) [inline datastore/elastic_search/query.py:603]: E203 whitespace before ':'

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Review by tanaya-b — APPROVED** [review]

**Pull request merged**